controllers/cache_handler.go

```diff
@@ -511,7 +511,13 @@ func (r *DisruptionReconciler) manageInstanceSelectorCache(instance *chaosv1beta
 
 		// start the cache with a cancelable context and duration, and attach it to the controller as a watch source
 		ch := make(chan error)
-		cacheCtx, cacheCancelFunc := context.WithTimeout(context.Background(), instance.Spec.Duration.Duration()+*r.ExpiredDisruptionGCDelay*2)
+		deletionDelay := time.Minute * 2
+
+		if r.ExpiredDisruptionGCDelay != nil {
```

Comment:
If `ExpiredDisruptioonGCDelay` is not set and it [defaults to `-1`](https://github.com/DataDog/chaos-controller/blob/d405aa7935475f82b8896151eb338f12b6f1b11a/main.go#L152-L153) will this have the desired effect?

Reply:
yes, the value is set to an empty pointer [there](https://github.com/DataDog/chaos-controller/blob/d405aa7935475f82b8896151eb338f12b6f1b11a/main.go#L354). it's what created the segfault